internal/testsacc/vdc_datasource_test.go

```diff
@@ -2,44 +2,77 @@
 package testsacc
 
 import (
+	"context"
 	"regexp"
 	"testing"
 
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
 
+	"github.com/orange-cloudavenue/terraform-provider-cloudavenue/internal/helpers/testsacc"
 	"github.com/orange-cloudavenue/terraform-provider-cloudavenue/pkg/uuid"
 )
 
-const testAccVDCDataSourceConfig = `
-data "cloudavenue_vdc" "test" {
-	name = "MyVDC"
+var _ testsacc.TestACC = &VDCDataSource{}
+
+const (
+	VDCDataSourceName = testsacc.ResourceName("data.cloudavenue_vdc")
+)
+
+type VDCDataSource struct{}
+
+func NewVDCDataSourceTest() testsacc.TestACC {
+	return &VDCDataSource{}
+}
+
+// GetResourceName returns the name of the resource.
+func (r *VDCDataSource) GetResourceName() string {
+	return VDCDataSourceName.String()
 }
-`
 
-func TestAccVDCDataSource(t *testing.T) {
-	dataSourceName := "data.cloudavenue_vdc.test"
+func (r *VDCDataSource) DependenciesConfig() (configs testsacc.TFData) {
+	configs.Append(GetResourceConfig()[VDCResourceName]().GetDefaultConfig())
+	return
+}
+
+func (r *VDCDataSource) Tests(ctx context.Context) map[testsacc.TestName]func(ctx context.Context, resourceName string) testsacc.Test {
+	return map[testsacc.TestName]func(ctx context.Context, resourceName string) testsacc.Test{
+		// * Test One (example)
+		"example": func(_ context.Context, resourceName string) testsacc.Test {
+			return testsacc.Test{
+				// ! Create testing
+				Create: testsacc.TFConfig{
+					TFConfig: `
+					data "cloudavenue_vdc" "example" {
+						name = cloudavenue_vdc.example.name
+					}`,
+					// TFConfig: testsacc.GenerateFromTemplate("cloudavenue_vdc",`
+					// data "cloudavenue_vdc" "example" {
+					// 	name = {{ get . "name" }}
+					// }`,
+					// Checks: NewVDCResourceTest().Tests(ctx)["example"](ctx, resourceName).GenerateCheckWithCommonChecks(),
```

Comment:
```suggestion
					Checks: NewVDCResourceTest().Tests(ctx)["example"](ctx, resourceName).GenerateCheckWithCommonChecks(),
```

Use this for using the same test in resource with resourceName in Datasource

Reply:
unable to use due to a problem to get value from template in dependencies
```
 TF_ACC=1 go test  -v ./internal/testsacc/ -count=1 -timeout 30m -run TestVDCDataSource
=== RUN   TestVDCDataSource
    vdc_datasource_test.go:71: Step 1/1 error: Check failed: 2 errors occurred:
                * Check 9/13 error: data.cloudavenue_vdc.example: Attribute 'name' expected "QBWc9h64pGSvazlv", got "C3DZjzXsX2GaIyR0"
                * Check 10/13 error: data.cloudavenue_vdc.example: Attribute 'description' expected "Etiamne flete hi intuerer.", got "Pro conexos meo."
        
--- FAIL: TestVDCDataSource (57.62s)
FAIL
FAIL    github.com/orange-cloudavenue/terraform-provider-cloudavenue/internal/testsacc  57.988s
FAIL
```